pkg/minishift/docker/image/image_handler.go

```diff
@@ -0,0 +1,348 @@
+	err = session.Run(cmd)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Error running command '%s': %v", cmd, err))
+	}
+
+	return nil
+}
+
+// availableImages returns a slice of available images in the current Docker instance
+func (handler *DockerImageHandler) availableImages() ([]string, error) {
+	session, err := handler.createSshSession()
+	if err != nil {
+		return nil, err
+	}
+	defer session.Close()
+
+	cmd := "docker images --format '{{.Repository}}:{{.Tag}}'"
+	var buffer bytes.Buffer
+	session.Stdout = &buffer
+	err = session.Run(cmd)
+	if err != nil {
+		return nil, errors.New(fmt.Sprintf("Error running command '%s': %v", cmd, err))
+	}
+
+	availableImages := strings.Split(buffer.String(), "\n")
+	return availableImages, nil
+}
+
+// imageNameToFileName maps an image name in the format repository/name:tag to a file system name.
+func (handler *DockerImageHandler) imageNameToFileName(image string) string {
+	imageNameParts := strings.FieldsFunc(image, func(r rune) bool {
+		switch r {
+		case '/', ':':
+			return true
+		}
+		return false
+	})
+
+	return strings.Join(imageNameParts, "@")
+}
+
+// isImageFile returns true if the denoted path represents a image file (based on file name).
+func (handler *DockerImageHandler) isImageFile(path string) bool {
+	fileName := filepath.Base(path)
+	match, err := regexp.Match("^.*@.*@.*$", []byte(fileName))
+	if err != nil {
+		return false
+	}
+	return match
+}
+
+// fileNameToImageName maps a filename on the host to an image name in the format repository/name:tag.
+func (handler *DockerImageHandler) fileNameToImageName(path string) string {
+	fileName := filepath.Base(path)
+	parts := strings.Split(fileName, "@")
+
+	return parts[0] + "/" + parts[1] + ":" + parts[2]
+}
+
+// isCached returns true if the specified image (file) exists on the host.
+func (handler *DockerImageHandler) isCached(imageCacheFile string) bool {
+	return filehelper.Exists(imageCacheFile)
+}
+
+// createSshSession creates an interactive SSH session
+func (handler *DockerImageHandler) createSshSession() (*ssh.Session, error) {
+	sshClient, err := sshutil.NewSSHClient(handler.driver)
+	if err != nil {
+		return nil, err
+	}
+
+	session, err := sshClient.NewSession()
+	if err != nil {
+		return nil, err
+	}
+
+	return session, nil
+}
+
+// IsPulled returns true is the specified image is already cached in the Docker daemon, false otherwise.
+func (handler *DockerImageHandler) IsPulled(image string) (bool, error) {
+	session, err := handler.createSshSession()
+	if err != nil {
+		return false, err
+	}
+	defer session.Close()
+
+	cmd := fmt.Sprintf("docker images -q %s", image)
+	var buffer bytes.Buffer
+	session.Stdout = &buffer
+	err = session.Run(cmd)
+	if err != nil {
+		return false, errors.New(fmt.Sprintf("Error running command '%s': %v", cmd, err))
+	}
+
+	if len(buffer.String()) > 0 {
+		return true, nil
+	} else {
+		return false, nil
+	}
+}
+
+func timeTrack(start time.Time, w io.Writer) {
```

Comment:
not specific to images export. I have added something before as part of debug around oc cluster up. I believe we should add more of this as output to the console.